main.cpp

```diff
@@ -131,41 +202,190 @@ std::pair<JSONObject, size_t> parse(std::string_view json) {
         }
         return {JSONObject{std::move(res)}, i};
     } else if (json[0] == '{') {
+        //for dict
         std::unordered_map<std::string, JSONObject> res;
         size_t i;
         for (i = 1; i < json.size();) {
             if (json[i] == '}') {
                 i += 1;
                 break;
             }
-            auto [keyobj, keyeaten] = parse(json.substr(i));
-            if (keyeaten == 0) {
-                i = 0;
-                break;
-            }
-            i += keyeaten;
-            if (!std::holds_alternative<std::string>(keyobj.inner)) {
-                i = 0;
-                break;
-            }
-            if (json[i] == ':') {
-                i += 1;
+            std::string key;
+            {//for key
+                std::regex elem_re(R"([ \n\r\t\v\f\0]*?[\'\"]{0,1}[a-zA-Z]+[\'\"]{0,1}[ \n\r\t\v\f\0]*?:)");
+                std::cmatch match;
+                std::regex_search(json.data() + i, json.data() + json.size(), match, elem_re);
+                key = match.str();
+                i += key.size();
+                key = key.substr(0, key.size() - 1);
+                int beg = 0;
+                while(std::string(" \n\r\t\v\f\0").find(key.at(beg)) != std::string::npos){
+                    beg++;
+                }
+                int end = key.size() - 1;
+                while(std::string(" \n\r\t\v\f\0").find(key.at(end)) != std::string::npos){
+                    end--;
+                }
+
+                key = key.substr(beg, end - beg + 1);
             }
-            std::string key = std::move(std::get<std::string>(keyobj.inner));
             auto [valobj, valeaten] = parse(json.substr(i));
             if (valeaten == 0) {
                 i = 0;
                 break;
             }
             i += valeaten;
             res.try_emplace(std::move(key), std::move(valobj));
+            //solve issue : can't allow any space behind value
+            while(std::string(" \n\r\t\v\f\0").find(json[i])!=std::string::npos){
+                i++;
+            }
             if (json[i] == ',') {
                 i += 1;
             }
         }
         return {JSONObject{std::move(res)}, i};
     }
-    return {JSONObject{std::nullptr_t{}}, 0};
+    else if(json[0]=='t'){
+        //for bool
+        if(auto index = json.find("true"); index==0){
```

Comment:
This is $O(N)$ if json is "t,,,,,,,,,,,,,,,,,,...", use C++20 `json.starts_with("true")` or `json.substr(0, 4) == "true"`